diff --git a/create-i18n/index.js b/create-i18n/index.js
index 8da0260a3bce1b4fd83aef7dc5ec48b8c96c4bfe..2df510eef575df81ef9657c05ff23d6e37069598 100644
--- a/create-i18n/index.js
+++ b/create-i18n/index.js
@@ -67,7 +67,7 @@ export function createI18n(locale, opts) {
       t.component = componentName
       if (define.cache[baseLocale][componentName]) {
         let isHMR = import.meta && (import.meta.hot || import.meta.webpackHot)
-        if (isHMR) {
+        if (isHMR && !opts.isSSR) {
           /* c8 ignore next 3 */
           for (let i in define.cache) {
             delete define.cache[i][componentName]
